test flaky test_quorum_inserts/test.py::test_drop_replica_and_achieve_quorum

https://s3.amazonaws.com/clickhouse-test-reports/43516/700181ca2e256ae477477b9f5ad0be5e2cab92d2/integration_tests__release__[2/4].html

<details>
<summary> Log example </summary>

```
        assert TSV("1\t2011-01-01\n2\t2012-02-02\n") == TSV(
            first.query("SELECT * FROM test_drop_replica_and_achieve_quorum ORDER BY a")
        )
>       assert TSV("1\t2011-01-01\n2\t2012-02-02\n") == TSV(
            second.query("SELECT * FROM test_drop_replica_and_achieve_quorum ORDER BY a")
        )

test_quorum_inserts/test.py:158: 
_ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ 
helpers/cluster.py:3059: in query
    return self.client.query(
helpers/client.py:36: in wrap
    return func(self, *args, **kwargs)
helpers/client.py:63: in query
    return self.get_query_request(
_ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ 

self = <helpers.client.CommandRequest object at 0x7ffa58441ac0>

    def get_answer(self):
        self.process.wait(timeout=DEFAULT_QUERY_TIMEOUT)
        self.stdout_file.seek(0)
        self.stderr_file.seek(0)
    
        stdout = self.stdout_file.read().decode("utf-8", errors="replace")
        stderr = self.stderr_file.read().decode("utf-8", errors="replace")
    
        if (
            self.timer is not None
            and not self.process_finished_before_timeout
            and not self.ignore_error
        ):
            logging.debug(f"Timed out. Last stdout:{stdout}, stderr:{stderr}")
            raise QueryTimeoutExceedException("Client timed out!")
    
        if (self.process.returncode != 0 or stderr) and not self.ignore_error:
>           raise QueryRuntimeException(
                "Client failed! Return code: {}, stderr: {}".format(
                    self.process.returncode, stderr
                ),
                self.process.returncode,
                stderr,
            )
E           helpers.client.QueryRuntimeException: Client failed! Return code: 33, stderr: Received exception from server (version 22.12.1):
E           Code: 289. DB::Exception: Received from 172.16.5.5:9000. DB::Exception: Replica doesn't have part '20120202_0_0_0' which was successfully written to quorum of other replicas. Send query to another replica or disable 'select_sequential_consistency' setting. Stack trace:
E           
E           0. DB::Exception::Exception(DB::Exception::MessageMasked const&, int, bool) @ 0xe46789a in /usr/bin/clickhouse
E           1. ? @ 0x8291d20 in /usr/bin/clickhouse
E           2. DB::StorageReplicatedMergeTree::getMaxAddedBlocks() const @ 0x145dc7eb in /usr/bin/clickhouse
E           3. DB::MergeTreeData::getQueryProcessingStageWithAggregateProjection(std::__1::shared_ptr<DB::Context const>, std::__1::shared_ptr<DB::StorageSnapshot> const&, DB::SelectQueryInfo&) const @ 0x14975ba2 in /usr/bin/clickhouse
E           4. DB::MergeTreeData::getQueryProcessingStage(std::__1::shared_ptr<DB::Context const>, DB::QueryProcessingStage::Enum, std::__1::shared_ptr<DB::StorageSnapshot> const&, DB::SelectQueryInfo&) const @ 0x14979f91 in /usr/bin/clickhouse
E           5. DB::InterpreterSelectQuery::getSampleBlockImpl() @ 0x13e2d931 in /usr/bin/clickhouse
E           6. ? @ 0x13e26295 in /usr/bin/clickhouse
E           7. DB::InterpreterSelectQuery::InterpreterSelectQuery(std::__1::shared_ptr<DB::IAST> const&, std::__1::shared_ptr<DB::Context> const&, std::__1::optional<DB::Pipe>, std::__1::shared_ptr<DB::IStorage> const&, DB::SelectQueryOptions const&, std::__1::vector<std::__1::basic_string<char, std::__1::char_traits<char>, std::__1::allocator<char>>, std::__1::allocator<std::__1::basic_string<char, std::__1::char_traits<char>, std::__1::allocator<char>>>> const&, std::__1::shared_ptr<DB::StorageInMemoryMetadata const> const&, std::__1::shared_ptr<DB::PreparedSets>) @ 0x13e210a9 in /usr/bin/clickhouse
E           8. DB::InterpreterSelectWithUnionQuery::buildCurrentChildInterpreter(std::__1::shared_ptr<DB::IAST> const&, std::__1::vector<std::__1::basic_string<char, std::__1::char_traits<char>, std::__1::allocator<char>>, std::__1::allocator<std::__1::basic_string<char, std::__1::char_traits<char>, std::__1::allocator<char>>>> const&) @ 0x13ebef42 in /usr/bin/clickhouse
E           9. DB::InterpreterSelectWithUnionQuery::InterpreterSelectWithUnionQuery(std::__1::shared_ptr<DB::IAST> const&, std::__1::shared_ptr<DB::Context>, DB::SelectQueryOptions const&, std::__1::vector<std::__1::basic_string<char, std::__1::char_traits<char>, std::__1::allocator<char>>, std::__1::allocator<std::__1::basic_string<char, std::__1::char_traits<char>, std::__1::allocator<char>>>> const&) @ 0x13ebc6e2 in /usr/bin/clickhouse
E           10. DB::InterpreterFactory::get(std::__1::shared_ptr<DB::IAST>&, std::__1::shared_ptr<DB::Context>, DB::SelectQueryOptions const&) @ 0x13de04de in /usr/bin/clickhouse
E           11. ? @ 0x141bf560 in /usr/bin/clickhouse
E           12. DB::executeQuery(std::__1::basic_string<char, std::__1::char_traits<char>, std::__1::allocator<char>> const&, std::__1::shared_ptr<DB::Context>, bool, DB::QueryProcessingStage::Enum) @ 0x141bce4d in /usr/bin/clickhouse
E           13. DB::TCPHandler::runImpl() @ 0x14e747ce in /usr/bin/clickhouse
E           14. DB::TCPHandler::run() @ 0x14e87cd9 in /usr/bin/clickhouse
E           15. Poco::Net::TCPServerConnection::start() @ 0x17cdd2f4 in /usr/bin/clickhouse
E           16. Poco::Net::TCPServerDispatcher::run() @ 0x17cdec9b in /usr/bin/clickhouse
E           17. Poco::PooledThread::run() @ 0x17e6daa7 in /usr/bin/clickhouse
E           18. Poco::ThreadImpl::runnableEntry(void*) @ 0x17e6b4dd in /usr/bin/clickhouse
E           19. ? @ 0x7f1c5d7b5609 in ?
E           20. clone @ 0x7f1c5d6da133 in ?
E           . (REPLICA_IS_NOT_IN_QUORUM)
E           (query: SELECT * FROM test_drop_replica_and_achieve_quorum ORDER BY a)
```

</details>

Flaky stats:
https://play.clickhouse.com/play?user=play#c2VsZWN0IAp0b1N0YXJ0T2ZIb3VyKGNoZWNrX3N0YXJ0X3RpbWUpIGFzIGQsCmNvdW50KCksICBncm91cFVuaXFBcnJheShwdWxsX3JlcXVlc3RfbnVtYmVyKSwgIGFueShyZXBvcnRfdXJsKQpmcm9tIGNoZWNrcyB3aGVyZSAnMjAyMi0wNi0wMScgPD0gY2hlY2tfc3RhcnRfdGltZSBhbmQgdGVzdF9uYW1lIGxpa2UgJyV0ZXN0X2Ryb3BfcmVwbGljYV9hbmRfYWNoaWV2ZV9xdW9ydW0lJyBhbmQgdGVzdF9zdGF0dXMgaW4gKCdGQUlMJywgJ0ZMQUtZJykgZ3JvdXAgYnkgZCBvcmRlciBieSBkIGRlc2M=

There is even [a crash a few days ago](https://s3.amazonaws.com/clickhouse-test-reports/48267/356716f0a1d510d7331e9cb7cbba3eed763050ef/integration_tests__asan__[4/6]/integration_run_parallel2_0.log)
```
E           helpers.client.QueryRuntimeException: Client failed! Return code: 32, stderr: [zero] 2023.03.31 05:50:40.740728 [ 285 ] <Fatal> BaseDaemon: ########################################
E           [zero] 2023.03.31 05:50:40.741188 [ 285 ] <Fatal> BaseDaemon: (version 23.3.1.1, build id: 7E6C686E3C47DF7403C2362B3E8989395BC5AEC0) (from thread 274) (query_id: b60c4b46-31cd-47c1-b07a-d8318ee95d6b) (query: DROP TABLE IF EXISTS test_drop_replica_and_achieve_quorum ON CLUSTER cluster) Received signal sanitizer trap (-3)
E           [zero] 2023.03.31 05:50:40.741376 [ 285 ] <Fatal> BaseDaemon: Sanitizer trap.
E           [zero] 2023.03.31 05:50:40.741536 [ 285 ] <Fatal> BaseDaemon: Stack trace: 0x202d68f7 0x2092811a 0xed00126 0xece2fa7 0xece5dbc 0xece6a48 0x2f93f339 0x3391a1bf 0x363019f9 0x36300ce5 0x36348694 0x36327b3d 0x363264c6 0x36325eae 0x36352c18 0x204e4cef 0x204f17a0 0x7f2083fdb609 0x7f2083f00133
E           [zero] 2023.03.31 05:50:40.824396 [ 285 ] <Fatal> BaseDaemon: 0. ./build_docker/./src/Common/StackTrace.cpp:287: StackTrace::tryCapture() @ 0x202d68f7 in /usr/bin/clickhouse
E           [zero] 2023.03.31 05:50:40.985744 [ 285 ] <Fatal> BaseDaemon: 1. ./build_docker/./src/Daemon/BaseDaemon.cpp:441: sanitizerDeathCallback() @ 0x2092811a in /usr/bin/clickhouse
E           [zero] 2023.03.31 05:50:47.401844 [ 285 ] <Fatal> BaseDaemon: 2. __sanitizer::Die() @ 0xed00126 in /usr/bin/clickhouse
E           [zero] 2023.03.31 05:50:53.653944 [ 285 ] <Fatal> BaseDaemon: 3. ? @ 0xece2fa7 in /usr/bin/clickhouse
E           [zero] 2023.03.31 05:50:59.920209 [ 285 ] <Fatal> BaseDaemon: 4. __asan::ReportGenericError(unsigned long, unsigned long, unsigned long, unsigned long, bool, unsigned long, unsigned int, bool) @ 0xece5dbc in /usr/bin/clickhouse
E           Error on processing query: Code: 32. DB::Exception: Attempt to read after eof: while receiving packet from 172.16.10.5:9000. (ATTEMPT_TO_READ_AFTER_EOF), Stack trace (when copying this message, always include the lines below):
E           
```

--- a/tests/integration/test_quorum_inserts/test.py
+++ b/tests/integration/test_quorum_inserts/test.py
@@ -144,25 +144,6 @@ def test_drop_replica_and_achieve_quorum(started_cluster):
         )
     )
 
-    print("Now we can insert some other data.")
-    zero.query(
-        "INSERT INTO test_drop_replica_and_achieve_quorum(a,d) VALUES (2, '2012-02-02')"
-    )
-
-    assert TSV("1\t2011-01-01\n2\t2012-02-02\n") == TSV(
-        zero.query("SELECT * FROM test_drop_replica_and_achieve_quorum ORDER BY a")
-    )
-    assert TSV("1\t2011-01-01\n2\t2012-02-02\n") == TSV(
-        first.query("SELECT * FROM test_drop_replica_and_achieve_quorum ORDER BY a")
-    )
-    assert TSV("1\t2011-01-01\n2\t2012-02-02\n") == TSV(
-        second.query("SELECT * FROM test_drop_replica_and_achieve_quorum ORDER BY a")
-    )
-
-    zero.query(
-        "DROP TABLE IF EXISTS test_drop_replica_and_achieve_quorum ON CLUSTER cluster"
-    )
-
 
 @pytest.mark.parametrize(("add_new_data"), [False, True])
 def test_insert_quorum_with_drop_partition(started_cluster, add_new_data):

